src/jit/gentree.cpp

```diff
@@ -17422,6 +17422,16 @@ bool GenTree::isRMWHWIntrinsic(Compiler* comp)
     switch (AsHWIntrinsic()->gtHWIntrinsicId)
     {
         case NI_SSE42_Crc32:
+        case NI_FMA_MultiplyAdd:
```

Comment:
It seems to me that perhaps we should just use a flag for this - especially if there are more intrinsics that will have this behavior. As I think I've mentioned before I don't like negative booleans - so I would prefer changing the existing flag to something like `HW_Flag_SSE_RMWSemantics` and then adding `HW_Flag_RMWSemantics`. But that is a topic for another discussion.

Reply:
> It seems to me that perhaps we should just use a flag for this - especially if there are more intrinsics that will have this behavior.

I don't think we have that many of them (there are very few RMW intrinsics for the VEX encoding). I'll add a comment indicating we should revisit if this grows any more.

> As I think I've mentioned before I don't like negative booleans - so I would prefer changing the existing flag to something like HW_Flag_SSE_RMWSemantics and then adding HW_Flag_RMWSemantics. But that is a topic for another discussion.

Right, I think we have an issue tracking this. -- I think the primary problem right now is that the majority case is positive, and it is easier to track/add the minority case (whether positive or negative).

Reply:
Added a `TODO-XArch-Cleanup` comment here.